src/tensor/mod.rs

```diff
@@ -227,7 +227,7 @@ impl Tensor {
     /// [N1, ..., Nk, labels]. The returned tensor uses float values.
     /// Elements of the input vector are expected to be between 0 and labels-1.
     pub fn onehot(&self, labels: i64) -> Tensor {
-        Tensor::zeros([self.size(), vec![labels]].concat(), crate::wrappers::kind::FLOAT_CPU)
+        Tensor::zeros([self.size(), vec![labels]].concat(), (self.kind(), self.device()))
```

Comment:
I think this function is supposed to return a float rather than the kind of the input tensor (see the documentation).

Reply:
I found documentation [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.one_hot.html) that says it should return a tensor of longs. Interestingly, `onehot()` is not listed in the `nn::functional` documentation [here](https://pytorch.org/docs/stable/nn.functional.html), nor found by searching from within the site. 

In this case, however, the `onehot()` function is defined on tensor, so I would expect the resulting onehot tensor to be the same type as the tensor it is applied to.

Reply:
I would rather keep the old behavior for now as it's a guarantee of the current api and just change the device as for this one it seems pretty clear that it should be preserved (I would have to think more about it but I don't see a scenario where one would want to keep an integer type for the onehot encoded - hence taking more time to think about this/gather potential users feedback).

Reply:
Ok, done. Restored the Float kind. Now the only change to the test is to confirm that the self and output tensors are on the same device.